### PR TITLE
Use onError to emit duplicate child key errors.

### PR DESCRIPTION
**Pull request by Laimiux**
## What
Use `onError` for duplicate child key errors instead of a custom function.

### CODE REVIEW

File: formula/src/main/java/com/instacart/formula/FormulaPlugins.kt [hunk 1]
```diff
@@ -30,14 +30,6 @@ object FormulaPlugins {
         plugin?.onError(error)
     }
 
-    fun onDuplicateChildKey(
```
Comment (Laimiux) [inline formula/src/main/java/com/instacart/formula/FormulaPlugins.kt:33]: Removing this in favor of using `FormulaPlugins.onError` with `FormulaError`

File: formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt [hunk 2]
```diff
@@ -814,7 +815,8 @@ class FormulaRuntimeTest {
             }
         }
 
-        parentFormula.test().input(Unit).output {
+        // Need failOnError to avoid test failing due to duplicate child errors
+        parentFormula.test(failOnError = false).input(Unit).output {
```
Comment (Laimiux) [inline formula/src/test/java/com/instacart/formula/FormulaRuntimeTest.kt:819]: Previously, duplicate child errors were not emitted in tests.
